src/ui.rs

```diff
@@ -1309,6 +1334,34 @@ impl<'state, 'input> Recorder<'state, 'input> {
         }
     }
 
+    // Returns the previous or next SelectionKey of the same kind as the current
+    // selection key. If there are no other keys of the same kind, the current
+    // key is returned instead. If `select_previous` is true, the previous key
+    // is returned. Otherwise, the next key is returned.
+    fn select_prev_or_next_of_same_kind(&self, select_previous: bool) -> SelectionKey {
+        let (keys, index) = self.find_selection();
+        let iterate_keys_with_wrap_around = |i| -> Box<dyn DoubleEndedIterator<Item = _>> {
+            let forward_iter = keys[i + 1..] // Skip the current key
+                .iter()
+                .chain(keys[..i].iter());
+            if select_previous {
+                return Box::new(forward_iter.rev());
+            }
+            Box::new(forward_iter)
```

Comment:
```suggestion
            if select_previous {
                Box::new(forward_iter.rev())
            } else {
                Box::new(forward_iter)
            }
```

Reply:
Sounds good. I'll prefer this style going forward.